### DISCUSSION TITLE
Redirect URL issue after user sign up

### DISCUSSION THREAD

**Discussion by GiftChrisJava** [supabase/supabase#21092] [Questions]
I am using Supabase for authentication on my Next.js app deployed on Vercel. I have an issue with the redirect URL after the user signs up and clicks on the confirmation link in the email. The user is being redirected to http://localhost:3000/?code=f71b2549-8fa9-44f4-80b6-754caff5c410 instead of https://trackervideo.vercel.app/?code=f71b2549-8fa9-44f4-80b6-754caff5c410.

I have tried using the actual URL of my app on Vercel as the emailRedirectTo option in the supabase.auth.signUp function, as well as using environment variables to store the URL, but none of these worked. I have also cleared my browser cache and cookies, and tried different browsers and devices, but the issue persists.

I suspect that Supabase is still caching the localhost 3000 URL for the email redirect. Could you please help me clear the cache on your end, or suggest any other possible solutions?

my code...
my form..

export default function page() {
  return (
    <div >
      ....
      </div>
      <form action="../auth/signup" method="post">
        <input
          required
          type="text"
          name="email"
          id="email"
          placeholder="Enter Email"
          
        />
        <input
          required
          type="password"
          name="password"
          id="password"
          placeholder="Enter Password"
      
        />
        <div >
          <button >
            Sign Up
          </button>
          
        </div>
      </form>
    </div>
  );
}

my auth/signup
import { createRouteHandlerClient } from "@supabase/auth-helpers-nextjs";
import { cookies } from "next/headers";
import { NextRequest, NextResponse } from "next/server";

export async function POST(req) {
  const url = new URL(req.url);
  const cookieStore = cookies();

  const formData = await req.formData();

  const email = String(formData.get("email"));
  const password = String(formData.get("password"));

  const supabase = createRouteHandlerClient({ cookies: () => cookieStore });

  await supabase.auth.signUp({
    email,
    password,
    options: {
      emailRedirectTo: `https://trackervideo.vercel.app/auth/callback`,
    },
  });

  return NextResponse.redirect(`${url.origin}/message`, { status: 301 });
}

my auth/callback
import { createRouteHandlerClient } from "@supabase/auth-helpers-nextjs";
import { cookies } from "next/headers";
import { NextResponse } from "next/server";

export async function GET(req) {
  const url = new URL(req.url);
  const code = url.searchParams.get("code");

  if (code) {
    const cookieStore = cookies();
    const supabase = createRouteHandlerClient({ cookies: () => cookieStore });

    await supabase.auth.exchangeCodeForSession(code);
  }
  return NextResponse.redirect(new URL("/login", url.origin));

  // return NextResponse.redirect("/login");
}

**Comment by charislam** [comment]
Have you configured the Site URL in the dashboard? (Take a look at Point 2 here: https://supabase.com/docs/guides/auth/auth-email)

This is `localhost:3000` by default

**Comment by PiessensLander** [comment]
I'm having the same issue here after calling the `supabaseServiceRole.auth.admin.inviteUserByEmail()` function. All URLs are set correctly in my Supabase Dashboard. The email template respects the correct {{ .SiteUrl }} but the redirect after /auth/confirm to check the OTP redirects to localhost:300/{{whatever was set in the ?next parameter}}
